torchsnapshot/snapshot.py

```diff
@@ -212,8 +214,10 @@ def take(
             app_state=app_state,
             replicated=replicated or [],
         )
+        if storage_kwargs is None:
+            storage_kwargs = dict()
         storage = url_to_storage_plugin_in_event_loop(
-            url_path=path, event_loop=event_loop
+            url_path=path, event_loop=event_loop, **storage_kwargs,
```

Comment:
are there benfits of passing unpacking the dict here vs passing the whole dictionary as-is to the StoragePlugin backend? 

e.g.
```
class S3StoragePlugin(StoragePlugin):
    def __init__(self, root: str, **kwargs) -> None:
```
vs

```
class S3StoragePlugin(StoragePlugin):
    def __init__(self, root: str, storage_options: Dict[str, Any]) -> None:
```

Reply:
I also like dict better. We can unpack it once it reaches fsspec.

Reply:
Fixed.

But actually, I think `** storage_options` is better because we can keep plugins that do not take `storage_options`'s API not changed, like `FSStoragePlugin`